### ISSUE TITLE
cppia can't link extern abstract enum

### ISSUE THREAD

**Issue by ibilon** [HaxeFoundation/hxcpp#639]
Trying to use one from cppia results in an error like:
```
> ./host/Host script.cppia
 in Script::0x7fffd27d7cb0
   GetFieldByName at :0 main
Could not link static _Main.E_Impl_::N1 (3)
Error : Bad link
```

**n.hpp**
```
#pragma once

enum N {
	N1 = 7,
	N2 = -1 
};
```

**Host.hx**
```
@:enum @:unreflective extern abstract E (EImpl) {
	 @:native("N1") var E1;
	 @:native("N2") var E2;
}

@:include("./n.hpp") @:unreflective @:native("N") extern class EImpl {}

class Host {
	public static function main () {
		trace(E.E1);
		trace(E.E2);
	}
}
```

**Script.hx**
```
class Script {
	public static function main () {
		Host.main();
		trace(Host.E.E1);
		trace(Host.E.E2);
	}
}
```

Build the host `haxe -main cpp.cppia.Host -cpp host Host -D scriptable`, the script `haxe -main Script -cppia script.cppia` and run `./host/Host script.cppia`.
Using `-jit` doesn't change anything.

Sorry for the multi file example, couldn't figure out how to reduce to a single file.

**Comment by hughsando** [comment]
Yes, native externs are not going to work in cppia, so I'm going to need to
find some kind of solution.
I could potentially export the integers somehow.
One problem is that the enums are "unreflective" so cppia could have
trouble setting or interacting with them.  I think I will need to make
extern enums first-class citizens.

**Comment by ibilon** [comment]
I could try and remove the `@:unreflective`, but I remember having a lot of `can't convert Dynamic* to nativeType`.

In this sample it's fine without it (doesn't resolve the cppia error ofc).
